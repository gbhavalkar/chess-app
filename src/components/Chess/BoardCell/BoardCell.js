import React, {Component} from 'react'
import styled from 'styled-components'

const Cell = styled.div`
    border: black 2px solid;
    background-color: white;
    text-align: center;
    height: 4em;
`

Cell.displayName = "Cell"

class BoardCell extends Component{
    render(){
        return <Cell>{this.props.xPosition + ',' + this.props.yPosition}</Cell>
    }
}

export default BoardCell;