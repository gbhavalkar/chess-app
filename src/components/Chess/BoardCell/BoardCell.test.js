import React from 'react'
import BoardCell from './BoardCell'

describe('BoardCell Component', () => {

    it('renders without crashing', () => {
        const wrapper = shallow(<BoardCell xPosition="0" yPosition="0"/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('render the styled Cell', () => {
        const wrapper = shallow(<BoardCell xPosition="0" yPosition="0"/>)
        expect(wrapper.find('Cell')).toMatchSnapshot()
    })

    it('renders the comma separated position string', () => {
        const wrapper = shallow(<BoardCell xPosition="0" yPosition="0"/>)
        expect(wrapper.find('Cell').children().text()).toEqual("0,0")
    })
})