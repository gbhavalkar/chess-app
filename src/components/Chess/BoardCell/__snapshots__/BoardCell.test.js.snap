// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardCell Component render the styled Cell 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <BoardCell
      xPosition="0"
      yPosition="0"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "0,0",
      },
      "ref": null,
      "rendered": "0,0",
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "0,0",
        },
        "ref": null,
        "rendered": "0,0",
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromError": true,
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "0,0",
    },
    "ref": null,
    "rendered": "0,0",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "0,0",
      },
      "ref": null,
      "rendered": "0,0",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "0,0",
      },
      "ref": null,
      "rendered": "0,0",
      "type": [Function],
    },
  ],
}
`;

exports[`BoardCell Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardCell
    xPosition="0"
    yPosition="0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "0,0",
    },
    "ref": null,
    "rendered": "0,0",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "0,0",
      },
      "ref": null,
      "rendered": "0,0",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
